### PR TITLE
Storybook: WP Dashboard and WP Story Editor actions update

### PR DESCRIPTION
**Pull request by BrookeGraham**
## Context
We want to use Storybook's `actions` from the `essentials add-on` package. Removing the need for the `actions add-on` package.

## Summary

We now define `action` within the `argTypes` when declaring the story.

## Relevant Technical Choices
Removed the `actions add-on` devDep for the `WP-Dashboard` and `WP-Story Editor` packages since all stories in those packages are now using actions from the `essentials add-on`.

## To-do

See #10461 

## User-facing changes

n/a
## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Fire up Storybook
List of stories updated:
- Dashboard/Views/Shared/TelemetryBanner
- Dashboard/Views/EditorSettings/VideoCache
- Dashboard/Views/EditorSettings/Telemetry
- Dashboard/Views/EditorSettings/MediaOptimization
- Dashboard/Views/EditorSettings/CustomFonts
- Dashboard/Views/EditorSettings/AdManagement/GoogleAdSense
- Dashboard/Views/EditorSettings/AdManagement/GoogleAdManager

- Stories Editor/Components/Dialog/Status Check Failed
- Stories Editor/Components/Dialog/Post Take Over Dialog
- Stories Editor/Components/Dialog/Post Lock Dialog

### Does this PR have a security-related impact?
no

### Does this PR change what data or activity we track or use?
no

### Does this PR have a legal-related impact?
no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Partially addresses #10375

### CODE REVIEW

**Review by BrittanyIRL — APPROVED** [review]
Thanks for breaking these up! 😅

File: packages/wp-dashboard/src/components/editorSettings/customFonts/stories/index.js [hunk 1]
```diff
@@ -48,16 +51,19 @@ export const _default = () => {
...
-    setAddedFonts((currentFonts) => {
-      return currentFonts.filter(({ id }) => id !== toDelete);
-    });
-  }, []);
+      setAddedFonts((currentFonts) => {
+        return currentFonts.filter(({ id }) => id !== toDelete);
+      });
+    },
+    [args]
+  );
```
Comment (samwhale) [inline packages/wp-dashboard/src/components/editorSettings/customFonts/stories/index.js:66]: This is cool 👍

File: packages/wp-dashboard/src/components/editorSettings/adManagement/googleAdSense/stories/index.js [hunk 2]
```diff
@@ -30,21 +26,12 @@ export default {
...
-  return (
-    <GoogleAdSense
-      handleUpdatePublisherId={(newPublisherId) =>
-        action('update publisher id')(newPublisherId)
-      }
-      handleUpdateSlotId={(newSlotId) => action('update slot id')(newSlotId)}
-      {...args}
-    />
-  );
+  return <GoogleAdSense {...args} />;
```
Comment (samwhale) [inline packages/wp-dashboard/src/components/editorSettings/adManagement/googleAdSense/stories/index.js:36]: 🤔  Seems like there may be a bug in the ad sense component:

The formatter flags this as invalid if there's a space after it. Would be good to add a ticket for this to fix later!
image

Comment (samwhale) [reply packages/wp-dashboard/src/components/editorSettings/adManagement/googleAdSense/stories/index.js:36]: Same for the second input!!!
image

Comment (samwhale) [inline packages/wp-dashboard/src/components/editorSettings/adManagement/googleAdSense/stories/index.js:36]: Nit: would be good to add a container that adds some padding so people can see the components better (and the spacing around them)

Not related to this ticket though. I can add a small ticket for that.
image

File: packages/wp-dashboard/src/components/editorSettings/adManagement/googleAdManager/stories/index.js [hunk 3]
```diff
@@ -29,20 +25,11 @@ export default {
...
 export const _default = (args) => {
-  return (
-    <GoogleAdManager
-      handleUpdate={(newSlotId) =>
-        action('update google ad manager')(newSlotId)
-      }
-      {...args}
-    />
-  );
+  return <GoogleAdManager {...args} />;
```
Comment (samwhale) [inline packages/wp-dashboard/src/components/editorSettings/adManagement/googleAdManager/stories/index.js:34]: Same here: I'll add this to the same ticket
image

Comment (samwhale) [inline packages/wp-dashboard/src/components/editorSettings/adManagement/googleAdManager/stories/index.js:34]: Also adding a space before breaks this:
image

I'll add to a different ticket

File: packages/wp-story-editor/src/components/postLock/stories/postTakeOverDialog.js [hunk 4]
```diff
@@ -27,21 +22,26 @@ import PostTakeOverDialog from '../postTakeOverDialog';
...
+  },
+  parameters: {
+    controls: { exclude: ['user'] },
+  },
+};
+
+export const _default = (args) => {
+  const user = {
+    name: args.name,
+    avatar: args.avatar,
```
Comment (samwhale) [inline packages/wp-story-editor/src/components/postLock/stories/postTakeOverDialog.js]: 💭  should `name` and `avatar` be removed from the `args` object so they aren't passed into `PostTakeOverDialog`?

Comment (samwhale) [reply packages/wp-story-editor/src/components/postLock/stories/postTakeOverDialog.js]: Or maybe we could update args to be
```js
args: {
    user: {
        name: 'Matt Mullenweg',
        avatar: 'avatar-url'
    }
}
```

File: packages/wp-story-editor/src/components/postLock/stories/postLockDialog.js [hunk 5]
```diff
@@ -28,18 +23,26 @@ export default {
...
   component: PostLockDialog,
   args: {
-    user: {
-      name: 'Matt Mullenweg',
-      avatar:
-        'http://1.gravatar.com/avatar/767fc9c115a1b989744c755db47feb60?size=48',
-    },
+    name: 'Matt Mullenweg',
+    avatar:
+      'http://1.gravatar.com/avatar/767fc9c115a1b989744c755db47feb60?size=48',
```
Comment (samwhale) [inline packages/wp-story-editor/src/components/postLock/stories/postLockDialog.js]: same here

**Review by samwhale — APPROVED** [review]
yayaaaay

File: packages/wp-story-editor/src/components/postLock/stories/postTakeOverDialog.js (hunk 4 above)
Comment (BrookeGraham) [reply packages/wp-story-editor/src/components/postLock/stories/postTakeOverDialog.js]: Totally. I only broke it out cause I like the string input ui on controls over editing via an object. BUT the object is more true to form. 
Updated.
Screen Shot 2022-02-07 at 1 59 45 PM

**Pull request merged**